### PR TITLE
Define searching preferences. #9

### PR DESCRIPTION
**Pull request by OussamaJmili**
#9 Define searching preferences..
* User can now store his searching preferences to use later for job searching.
* User can now search for jobs too using the searching bar.
* User now has the choice to use his stored preferences or search for all jobs when searching.

### CODE REVIEW

File: linkedin-clone/src/components/Header.js [hunk 1]
```diff
@@ -1,19 +1,82 @@
...
 import { Link } from "react-router-dom";
 import { useState, useEffect } from "react";
+import { getAuth } from "firebase/auth";
+import { doc, getDoc } from "firebase/firestore";
+import db from '../firebase';
+import 'firebase/firestore';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog, faSearch } from '@fortawesome/free-solid-svg-icons';
+
+const getUserSearchingPreferences = async () => {
```
Comment (drbladz) [inline linkedin-clone/src/components/Header.js]: Move this function in index.js as we try to separate react components and utility functions that do not rely on props for cleaner code that is easier to define, read and test. This also avoids creating new instances with every re-render.

File: linkedin-clone/src/components/Header.js [hunk 2]
```diff
@@ -1,19 +1,82 @@
...
+      return null;
+    }
+  } catch (error) {
+    console.error('Error fetching user preferences:', error);
+    return null;
+  }
+};
+
+
+const filterJobsByPreferences = (jobs, preferences) => {
```
Comment (drbladz) [inline linkedin-clone/src/components/Header.js]: Same goes for this function.

**Review by nicotowa — APPROVED** [review]

**Review by drbladz — CHANGES_REQUESTED** [review]

File: linkedin-clone/src/components/Header.js (hunk 1 above)
Comment (OussamaJmili) [reply linkedin-clone/src/components/Header.js]: Done

File: linkedin-clone/src/components/Header.js (hunk 2 above)
Comment (OussamaJmili) [reply linkedin-clone/src/components/Header.js]: done

File: linkedin-clone/src/components/Header.js [hunk 3]
```diff
@@ -1,19 +1,50 @@
 import styled from "styled-components";
 import {connect } from "react-redux";
-import { signOutAPI, getUsers } from "../actions";
+import { signOutAPI, getUsers} from "../actions";
+import { getAllJobPostings } from "../actions/index";
 import { Link } from "react-router-dom";
 import { useState, useEffect } from "react";
+import { getAuth } from "firebase/auth";
```
Comment (drbladz) [inline linkedin-clone/src/components/Header.js]: please remove the unused imports

Comment (OussamaJmili) [reply linkedin-clone/src/components/Header.js]: Done

**Review by drbladz — APPROVED** [review]
Looks good

**Pull request merged**